modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/custom/filter/portlet/shared/search/CustomFilterPortletSharedSearchContributor.java

```diff
@@ -100,13 +98,12 @@ private String _getFilterValue(
 			return null;
 		}
 
-		Optional<String> parameterValueOptional =
-			portletSharedSearchSettings.getParameterOptional(
-				CustomFilterPortletUtil.getParameterName(
-					customFilterPortletPreferences));
+		String parameterValue = portletSharedSearchSettings.getParameter(
+			CustomFilterPortletUtil.getParameterName(
+				customFilterPortletPreferences));
 
-		if (parameterValueOptional.isPresent()) {
-			return parameterValueOptional.get();
+		if (Validator.isNotNull(parameterValue)) {
```

Comment:
@julschong It seems you are using `Validator.isNotNull` for string values in many places, while the original logic was just `Optional#isPresent/ifPresent`. Any specific reason?

Reply:
@dantewang parameterValue can be String or null in this case. I could also use `if (parameterValue == null)` as well. I've been using Validator.isNotNull to check for blank or null in this module. I thought to keep it consistent. 
What do you think work better?

Reply:
@julschong My concern is that `Optional#isPresent/ifPresent` only checks `null`, so the logic becomes more strict when replacing them with `Validator.isNotNull`. This conversion is not identical before/after.

That's why I want to know if there's any reason, maybe related to their business logic, that makes `Validator` a better choice.

Reply:
@dantewang I see. Thanks for the explaination :)
Validator usages are mostly checking the values gotten from PortletSharedSearchSettings.
The values are only going to be non-blank String value or null.
Validator.isNull, isNotNull check for 3 kinds: null, blank string, "null" string
the only new restriction here is "null" string. If you'd like me to switch to use == null, please let me know.